tests/Unit/NumericalAlgorithms/LinearOperators/Test_WeakDivergence.cpp

```diff
@@ -0,0 +1,365 @@
+                                                      [collapsed_index],
+                  local_approx);
+            }
+          }
+        }
+      }
+    } else {
+      for (size_t storage_index = 0;
+           storage_index < get<tag>(divergence_result).size() and
+           (Dim == 1 or mesh.extents(0) == 3);
+           ++storage_index) {
+        size_t collapsed_index = std::numeric_limits<size_t>::max();
+        if constexpr (Dim == 3) {
+          collapsed_index = ::collapsed_index(
+              Index<3>{mesh.extents(0) / 2, mesh.extents(1) / 2,
+                       mesh.extents(2) / 2},
+              mesh.extents());
+        } else if constexpr (Dim == 2) {
+          collapsed_index = ::collapsed_index(
+              Index<2>{mesh.extents(0) / 2, mesh.extents(1) / 2},
+              mesh.extents());
+        } else {
+          collapsed_index =
+              ::collapsed_index(Index<1>{mesh.extents(0) / 2}, mesh.extents());
+        }
+        CHECK_ITERABLE_CUSTOM_APPROX(
+            get<tag>(divergence_result)[storage_index][collapsed_index],
+            get<tag>(
+                expected_divergence_result)[storage_index][collapsed_index],
+            local_approx);
+      }
+    }
+  });
+}
+
+template <size_t Dim>
+void test_weak_divergence_constant_jacobian(const Mesh<Dim>& mesh) {
+  CAPTURE(Dim);
+  CAPTURE(mesh);
+  CAPTURE(mesh.quadrature(0));
+  std::uniform_real_distribution<double> dist(-1.0, 1.0);
+  MAKE_GENERATOR(gen);
+  using flux_tags =
+      tmpl::list<Tags::Flux<Var1, tmpl::size_t<Dim>, Frame::Inertial>,
+                 Tags::Flux<Var2<Dim>, tmpl::size_t<Dim>, Frame::Inertial>>;
+  using div_tags = tmpl::list<
+      Tags::div<Tags::Flux<Var1, tmpl::size_t<Dim>, Frame::Inertial>>,
+      Tags::div<Tags::Flux<Var2<Dim>, tmpl::size_t<Dim>, Frame::Inertial>>>;
+
+  const auto logical_coords = logical_coordinates(mesh);
+  tnsr::I<DataVector, Dim, Frame::Inertial> inertial_coords{
+      mesh.number_of_grid_points()};
+  Jacobian<DataVector, Dim, Frame::Logical, Frame::Inertial> jacobian{
+      mesh.number_of_grid_points(), 0.0};
+  InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>
+      inverse_jacobian{mesh.number_of_grid_points(), 0.0};
+  for (size_t i = 0; i < Dim; ++i) {
+    jacobian.get(i, i) = 2.0;
+    inertial_coords.get(i) = 2.0 * logical_coords.get(i);
+  }
+
+  InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>
+      det_jac_times_inverse_jacobian{};
+
+  dg::metric_identity_jacobian(make_not_null(&det_jac_times_inverse_jacobian),
+                               mesh, inertial_coords, jacobian);
+  const auto det_jacobian = determinant(jacobian);
+  for (size_t i = 0; i < Dim; ++i) {
+    for (size_t j = 0; j < Dim; ++j) {
+      inverse_jacobian.get(i, j) =
+          det_jac_times_inverse_jacobian.get(i, j) / get(det_jacobian);
+    }
+  }
+
+  // Generate constant fluxes that aren't all the same.
```

Comment:
Is this an out-of-date comment? AFAICT your fluxes here are non-constant (which I think is correct, given that the jacobians are constant)